contracts/libs/IbcUtils.sol

```diff
@@ -35,32 +35,35 @@ library IbcUtils {
      * hexStr is case-insensitive.
      */
     function hexStrToAddress(string memory hexStr) public pure returns (address addr) {
+        bytes memory hexBytes = bytes(hexStr);
         if (bytes(hexStr).length != 40) {
             revert IBCErrors.invalidHexStringLength();
         }
 
-        bytes memory strBytes = bytes(hexStr);
-        bytes memory addrBytes = new bytes(20);
-
-        for (uint256 i = 0; i < 20; i++) {
-            uint8 high = uint8(strBytes[i * 2]);
-            uint8 low = uint8(strBytes[1 + i * 2]);
-            // Convert to lowercase if the character is in uppercase
-            if (high >= 65 && high <= 90) {
-                high += 32;
-            }
-            if (low >= 65 && low <= 90) {
-                low += 32;
+        uint256 total = 0;
+        uint256 base = 1;
+        uint256 i = 40;
+        while (i > 0) {
+            i--;
+            uint256 digit;
+            // Convert ASCII to integer value
+            if (uint8(hexBytes[i]) >= 48 && uint8(hexBytes[i]) <= 57) {
+                // '0' to '9'
+                digit = uint160(uint8(hexBytes[i]) - 48);
+            } else if (uint8(hexBytes[i]) >= 65 && uint8(hexBytes[i]) <= 70) {
+                // 'A' to 'F'
+                digit = uint160(uint8(hexBytes[i]) - 55);
```

Comment:
please add a comment here explaining where this 55 and 87 below come from since it's not immediately obvious

Reply:
ah yes good idea, I added this